pkgs/checks/lib/src/extensions/string.dart

```diff
@@ -59,10 +59,18 @@ extension StringChecks on Subject<String> {
     );
   }
 
-  /// Expects that the string matches the regular expression [expected].
-  void matches(RegExp expected) {
+  /// Expects that the string matches the pattern [expected].
+  ///
+  /// Fails if [expected] returns an empty result from calling `allMatches` with
+  /// the value.
+  ///
+  /// ```
+  /// check(actual).matchesPattern('abc');
+  /// check(actual).matchesPattern(RegExp(r'\d'));
+  /// ```
+  void matchesPattern(Pattern expected) {
```

Comment:
Since people just loooove their regexps, consider _also_ providing a shorthand:
```dart
  void matchesRegExp(String source, {bool caseSensitive = true, multiline = false, unicode = false}) { 
      matchesPattern(RegExp(source, caseSensitive: caseSensitive, multiline: multiline, unicode: unicode);
  }
```

If you have a `RegExp` object already, you can just use `matchesPattern`, but it saves a little verbosity to not have to write the `RegExp(...)`.

(Or not, and wait for someone else to provide it.)

Reply:
Could `matchesPattern` return a `Subject<Match>`, so you can keep testing on the result, like:
```dart
  check(someString).matchesPattern(myRegExp)
     ..[0]!.contains("foo")
     ..[1].isNotNull.contains("foo");
```
(Is it better to return a `Subject<Match>?` or a `Subject<Match>` which may be inert/ignoring checks, when there is no match? I guess the latter is safer if you don't know immediately if there is a match.)